yoyodyne/models/pointer_generator.py

```diff
@@ -1,261 +1,75 @@
 """Pointer-generator model classes."""
 
+import math
 from typing import Optional, Tuple
 
 import torch
 from torch import nn
 
 from .. import batches
-from . import attention, generation_probability, lstm
+from . import lstm, modules
 
 
 class Error(Exception):
     pass
 
 
-class PointerGeneratorLSTMEncoderDecoderNoFeatures(lstm.LSTMEncoderDecoder):
-    """Pointer-generator model with an LSTM backend and no features.
+class GenerationProbability(nn.Module):
```

Comment:
I'm good with this here. But previously Kyle had requested it be moved into its own model file. @kylebgorman should this be a module? Or just keep it here in the model file?

Reply:
IMO positional encoding and generation probability are both complicated enoguh to be their own modules.

Reply:
I'm of the opinion that since they're small enough and only relevant to those respective modules that they might as well be included. (I may also just have an irrational dislike of modules imports...)